docroot/themes/contrib/civictheme/includes/views.inc

```diff
@@ -69,9 +97,46 @@ function civictheme_preprocess_views_view(&$variables) {
  * Implements hook_preprocess_HOOK().
  */
 function civictheme_preprocess_views_view_grid(&$variables) {
+  $view = $variables['view'] ?? NULL;
   // On Grid view title should be added to card container.
-  if ($variables['view']->getStyle()->getBaseId() === 'grid' && isset($variables['view']->header['result'])) {
+  if ($view->getStyle()->getBaseId() === 'grid' && isset($view->header['result'])) {
     $variables['modifier_class'] = 'civictheme-card-container--listing civictheme-card-container--with-background';
-    $variables['title'] = $variables['view']->header['result']->render();
+    $variables['title'] = $view->header['result']->render();
+  }
+  // Adds component theme settings to view / listing template.
+  $is_view_component = isset($variables['view']->civictheme_listing);
+  if ($is_view_component) {
+    $variables += $variables['view']->civictheme_listing;
+  }
+}
+
+/**
+ * Gets the Listing paragraph from route.
```

Comment:
Allows theme users to user this API to get access to listing component from ajax views - ie where we don't load paragraph on page load.